std/smtp/_helpers.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+/** Based on https://github.com/golang/go/blob/92c732e901a732855f4b813e6676264421eceae9/src/net/smtp/smtp.go
+  * Copyright 2010 The Go Authors. All rights reserved.
+  * Use of this source code is governed by a BSD-style
+  * license that can be found in the LICENSE file.
+  */
+export function validateLine(line: string): void {
+  if (line.includes("\r") || line.includes("\n")) {
+    throw new Error("smtp: A line must not contain CR or LF");
+  }
+}
+
+// TODO(uki00a) I'm not sure if std/smtp/mod.ts is the right place for this function.
+/** Adopted from https://github.com/golang/go/blob/92c732e901a732855f4b813e6676264421eceae9/src/net/ipsock.go
+ * Copyright 2009 The Go Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ *
+ * SplitHostPort splits a network address of the form "host:port",
+ * "host%zone:port", "[host]:port" or "[host%zone]:port" into host or
+ * host%zone and port.
+ * 
+ * A literal IPv6 address in hostport must be enclosed in square
+ * brackets, as in "[::1]:80", "[::1%lo0]:80".
+ * 
+ * See func Dial for a description of the hostport parameter, and host
+ * and port results.
+ */
+export function splitHostPort(hostport: string): [string, number] {
```

Comment:
Maybe move into _address_parameter_extrator.ts to remove the generically named helpers.ts?